prediction/src/views.py

```diff
@@ -81,29 +89,27 @@ def predict(algorithm):
         try:
             predict_method = PREDICTORS[algorithm]
 
-            if 'centroids' in payload:
+            if algorithm == 'identify':
+                prediction = predict_method(payload['dicom_path'])
+            else:
                 prediction = predict_method(payload['dicom_path'],
                                             payload['centroids'])
-            else:
-                prediction = predict_method(payload['dicom_path'])
 
             response.update({
                 'prediction': prediction
             })
 
-        except Exception:
-            error = "There was an error with your request."
-
-    # set the status code and messages for the response
-    if error:
-        response.update({
-            'error': error,
-            'status': 500
-        })
-    else:
-        response.update({
-            'status': 200,
-        })
+        except KeyError as e:
+            errormsg = 'Algorithm \'{}\' requires missing parameter {}'.format(algorithm, str(e))
+            return compose_error(errormsg)
+        except Exception as e:
+            # pass other errors from prediction function
+            return compose_error(str(e))
+
+    # set the status code for the response
+    response.update({
+        'status': 200
```

Comment:
Ah, that makes sense. I'd noticed those trailing commas but wasn't sure if they were intentional. I'll add them back in!